go/README.md

```diff
@@ -1 +1,177 @@
+# Go Code Conventions
+
+## Tooling
+
+- Visual Studio Code
+- delve
+- golangci-lint
+- staticcheck
+- modd
+
+## Modules
+
+go mod init bitbucket.org/aanzeeonline/[yourprojectname]
+
+Working with a local fork of a module, put an extra line in the go.mod:
+// replace bitbucket.org/aanzeeonline/libgoaz => ../LibGOAZ
+(and remove the comments)
+
+Before pushing the final update to git, make sure that the fork isn't in use anymore
+
+## project structure for an API / Web application
+
+main
+- start (explain why)
+- web
+- helpers
+- datamodels
+- business (separation of concerns)
+  - domains (if you have more than one)
+    - handlers
+    - queries (if you work with a database)
+    - repositories
+    - services
+    - validators
+
+## Makefile
+
+Every project must have a makefile.
+(You can look at other projects for an universal working version)
+
+## Package name
+
+A package name represents the content it houses. Inside there will be 1 or more files with functions that
+don't have the package name in front of it. Otherwise, when using a function, you get a stutter-effect,
+like this: database.DatabaseOpener(), while database.Opener() is enough.
+
+You can simplify a package name when it's a common abbreviation. database could be db, so db.Opener() can be used.
+
+When importing an external library (from aan zee), make it clear by using an alias, like:
+capErrors "bitbucket.org/aanzeeonline/capila/errors"
+libConv "bitbucket.org/aanzeeonline/libgoaz/conv"
+libCrypt "bitbucket.org/aanzeeonline/libgoaz/crypt"
+
+Don't use a name that it meaningless, like: util, common, misc.
+There is one exception to this: helpers.
+We will use helpers to store various helper-functions and over time, we can decide to group them into
+a package that is more suitable.
+
+## Imports
+
+order of imports
+
+- packages of current module
+- packages of aan zee lib's
+- the rest
+
+## Variable declaration
+
+Top level order of declaration:
+- constants
+- types
+- variables
+
+Global / package / function
+
+Don't use many Global var's.
+
+If a package var is used in more than 1 file, put it in a base.go
+
+Declare function var's mostly at the top of the function, or close to the first usage.
+
+## Functions and methods
+
+You can house more than 1 function inside a file (inside a package).
+The file name represents the content it houses.
+
+- naming
+	- give the function a name it represent.
+	- You can simplify a function name when it's a common abbreviation.
+		db.Initialize() could be db.Init()
+- no getters (the fact it begins with a capital and returns value, makes it a getter)
+- setters can be used
+- if crud, follow crud order inside source? (thoughts about this?)
+- global vs local
+  - first all global functions and methods (they start with capitals)
+  - local function below that (they start with lowercase)
+- return var's:
+  - when returning an error, it is always the last value that returns
+
+## Testing
+
+- code coverage (100% of relevant)
```

Comment:
Who decides what is relevant and how would one automate that decision?

Reply:
It is hard to start with, but i recommend 100% for all code

Reply:
check if go.1.13 has a solution for "skipping" coverage
For now: 100% until we decide different